lib/mutant/expression/method.rb

```diff
@@ -3,44 +3,44 @@ class Expression
 
     # Explicit method expression
     class Method < self
+      include Anima.new(:scope_name, :scope_symbol, :method_name)
+      private(*anima.attribute_names)
 
       MATCHERS = IceNine.deep_freeze(
         '.' => Matcher::Methods::Singleton,
         '#' => Matcher::Methods::Instance
       )
+      private_constant(*constants(false))
 
-      register(
-        /\A(?<scope_name>#{SCOPE_PATTERN})(?<scope_symbol>[.#])(?<method_name>#{METHOD_NAME_PATTERN})\z/
-      )
+      REGEXP = /\A(?<scope_name>#{SCOPE_PATTERN})(?<scope_symbol>[.#])(?<method_name>#{METHOD_NAME_PATTERN})\z/.freeze
```

Comment:
Assembling from variables may not be a bad idea though, since it looks like some parts of the regexp, such as `(?<scope_name>#{SCOPE_PATTERN})` are the same in a few subclasses.